### PR TITLE
[improvement](query) Improve fe high concurrent query performance

### LINKED ISSUE #7935: [Enhancement] Improve fe high concurrent query performance
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

Now, I will try to optimize the fe high concurrent query performance

image

### Solution

lower fe overhead

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by caiconghui**
# Proposed changes

Issue Number: close #7935 

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (No)
2. Has unit tests been added: (No)
3. Has document been added or modified: (No Need)
4. Does it need to update dependencies: (No Need)
5. Are there any changes that cannot be rolled back: (No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by caiconghui** [comment]
optimized statistic, 5k connection query 'select 1'
before 561ms 262ms 15.56k
after    406ms 130ms 32.2k

File: fe/fe-core/src/main/java/org/apache/doris/qe/ConnectScheduler.java [hunk 1]
```diff
@@ -96,50 +94,50 @@ public boolean submit(ConnectContext context) {
...
-            if (conns >= LdapAuthenticate.getMaxConn()) {
+            if (conns.incrementAndGet() > LdapAuthenticate.getMaxConn()) {
+                conns.decrementAndGet();
+                numberConnection.decrementAndGet();
                 return false;
             }
-        } else if (conns >= ctx.getCatalog().getAuth().getMaxConn(ctx.getQualifiedUser())) {
+        } else if (conns.incrementAndGet() > ctx.getCatalog().getAuth().getMaxConn(ctx.getQualifiedUser())) {
+            conns.decrementAndGet();
+            numberConnection.decrementAndGet();
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/qe/ConnectScheduler.java:113]: why call `numberConnection.decrementAndGet();` here?

File: fe/fe-core/src/main/java/org/apache/doris/qe/Coordinator.java [hunk 2]
```diff
@@ -471,8 +471,9 @@ public void exec() throws Exception {
                     addressToBackendID.get(execBeAddr),
                     toBrpcHost(execBeAddr),
                     queryOptions.query_timeout * 1000);
-
-            LOG.info("dispatch query job: {} to {}", DebugUtil.printId(queryId), topParams.instanceExecParams.get(0).host);
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("dispatch query job: {} to {}", DebugUtil.printId(queryId), topParams.instanceExecParams.get(0).host);
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/qe/Coordinator.java:475]: Actually this log is useful when tracking a query. Does it really impact the performance?

File: fe/fe-core/src/main/java/org/apache/doris/qe/ConnectScheduler.java (hunk 1 above)
Comment (caiconghui) [reply fe/fe-core/src/main/java/org/apache/doris/qe/ConnectScheduler.java:113]: > why call `numberConnection.decrementAndGet();` here?

we should reduce the connection number when connect failed

File: fe/fe-core/src/main/java/org/apache/doris/qe/Coordinator.java (hunk 2 above)
Comment (caiconghui) [reply fe/fe-core/src/main/java/org/apache/doris/qe/Coordinator.java:475]: by the flame graph and the statistic provided above，for most users，query performance is the most important thing

File: fe/fe-core/src/main/java/org/apache/doris/qe/ConnectScheduler.java (hunk 1 above)
Comment (morningman) [reply fe/fe-core/src/main/java/org/apache/doris/qe/ConnectScheduler.java:113]: Oh, I miss it.

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**